bitsandbytes/backends/cuda.py

```diff
@@ -0,0 +1,479 @@
+import ctypes as ct
+from typing import Optional, Tuple
+
+import torch
+
+from bitsandbytes.cextension import lib
+from bitsandbytes.functional import (
+    CUBLAS_Context,
+    coo_zeros,
+    dequantize_blockwise,
+    dtype2bytes,
+    get_4bit_type,
+    get_colrow_absmax,
+    get_ptr,
+    get_transform_buffer,
+    is_on_gpu,
+    post_call,
+    pre_call,
+    prod,
+    quantize_blockwise,
+)
+from bitsandbytes.utils import QuantState
+
+from .base import Backend
+
+
+class CUDABackend(Backend):
```

Comment:
Currently, other quantization functions for 8-bit optimizers and general k-bit quantization is missing.